### PR TITLE
fix: prevent user credentials to persist on login screen after logout & provide UI feedback on short password entry

### LINKED ISSUE #259: Lack of UI Feedback on Short Password Entry During Authentication
### 🐛 Describe the bug

**Description:**

This issue pertains to the authentication process of the application. When a user attempts to authenticate by entering a short password and pressing the login button, there is no feedback provided on the User Interface (UI) of the application. The only indication of an error is an exception that appears in the debug console.

**Steps to Reproduce:**

Navigate to the login page.
Enter a short password in the password field.
Press the login button.

**Expected Result:**

The application should provide feedback to the user, such as a snack bar notification or an alert, indicating that the entered password is too short.

**Actual Result:**

No feedback is provided on the UI. An exception appears in the debug console.

**Proposed Solution**

Implement a snack bar notification or an alert that informs the user when the entered password is too short. This will improve the user experience by providing clear and immediate feedback during the authentication process.

Screenshot_1708477568
Screenshot_1708477419

### LINKED ISSUE #260: User Credentials Persist on Login Screen After Logout
Description:

This issue pertains to the user authentication process in the application. After logging out from the account, the application redirects the user back to the login screen. However, the previously entered email and password are still visible in their respective fields. This poses a significant privacy concern as it allows anyone with access to the device to see the user’s credentials.

Steps to Reproduce:

Log into the application.
Navigate to the account settings and log out.
Observe the login screen after logout.
Expected Result:

The email and password fields should be cleared out when a user logs out, ensuring that the user’s credentials do not persist on the login screen.

Actual Result:

The user’s email and password remain visible on the login screen after logout.

### PR DESCRIPTION
**Pull request by AyaNady17**
## Description:
- When a user logs out of the app, their credentials (such as email and password) should not persist on the login screen. Currently, after logging out, the text fields still retain the user’s information, which is undesirable.

- When a user attempts to authenticate by entering a short password and pressing the login button, there is no feedback provided on the User Interface (UI) of the application. The only indication of an error is an exception that appears in the debug console.

## Solution: 
- Cleared Text Field Controllers: After a successful login, I ensured that the text field controllers (for email and password) are cleared. This prevents the previous user’s credentials from being displayed when the login screen is revisited.

- Implement a snack bar notification or an alert that informs the user when the entered password is too short.

Fixes #260 
Fixes #259 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

- first bug: I thoroughly tested the behavior by logging in, logging out, and verifying that the text fields are empty upon returning to the login screen.
- second bug: entered valid email with uncompleted password (less than 8 characters) and verify here is a Ui feedback 

 
##Before Fix
- first bug : 
https://github.com/AOSSIE-Org/Resonate/assets/144162711/5dcf6521-14d0-4f94-b2cc-cff0a99b7708
- second bug: 
https://github.com/AOSSIE-Org/Resonate/assets/144162711/189cc236-8670-4b89-a37d-e978fc90c64e

##After Fix 
- first bug : 
https://github.com/AOSSIE-Org/Resonate/assets/144162711/8f1e39f9-34d5-40cd-8de9-5107999f96ed
- second bug: 
https://github.com/AOSSIE-Org/Resonate/assets/144162711/a698fe11-9dfb-4182-96de-30929262f6c4

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Tag the PR with the appropriate labels

### CODE REVIEW

**Comment by AyaNady17** [comment]
@jddeep please review my code and inform me if I have to modify anything?

**Comment by AyaNady17** [comment]
@chandansgowda please review so I can edit any comments ASAP

File: lib/controllers/authentication_controller.dart [hunk 1]
```diff
@@ -30,7 +30,11 @@ class AuthenticationController extends GetxController {
     } on AppwriteException catch (e) {
       log(e.toString());
       if (e.type == 'user_invalid_credentials') {
-        customSnackbar('Try Again!', "Incorrect Email Or Password", MessageType.error);
+        customSnackbar(
+            'Try Again!', "Incorrect Email Or Password", MessageType.error);
+      } else if (e.type == 'general_argument_invalid') {
+        customSnackbar('Try Again!', "Password is less than 8 characters",
+            MessageType.error);
       }
```
Comment (jddeep) [inline lib/controllers/authentication_controller.dart:38]: please put the error types in constants file and use those constants.

**Review by jddeep — CHANGES_REQUESTED** [review]

File: lib/controllers/authentication_controller.dart (hunk 1 above)
Comment (AyaNady17) [reply lib/controllers/authentication_controller.dart]: @jddeep  I've modified that

Comment (AyaNady17) [reply lib/controllers/authentication_controller.dart]: any updates? @jddeep

**Comment by AyaNady17** [comment]
there was some conflicts and I've solved them, now the branch is ready to be merged @chandansgowda

**Pull request merged**